exts/cesium.omniverse.cpp.tests/cesium/omniverse/cpp/tests/extension.py

```diff
@@ -30,12 +31,21 @@ def on_startup(self):
         print("Started Cesium Tests Extension.")
 
     def run_once_after_stage_opens(self, _):
+        # wait until the USD stage is fully set up
         if omni.usd.get_context().get_stage_state() == omni.usd.StageState.OPENED:
-            print("Beginning Cesium Tests Extension tests")
-            stageId = omni.usd.get_context().get_stage_id()
-            tests_interface.run_all_tests(stageId)
-            print("Cesium Tests Extension tests complete")
-            self._run_once_sub.unsubscribe()
+            # set up tests on one frame, then run the tests on the next frame
+            # note we can't use wait_n_frames here as this is a subscribed function
+            # so it cannot be async
+            if not self.tests_set_up:
+                self.tests_set_up = True
+                print("Beginning Cesium Tests Extension tests")
+                stageId = omni.usd.get_context().get_stage_id()
+                tests_interface.set_up_tests(stageId)
+            else:
+                # unsubscribe so there's no way the next frame triggers another run
+                self._run_once_sub.unsubscribe()
+                tests_interface.run_all_tests()
+                print("Cesium Tests Extension tests complete")
```

Comment:
Funky. Could we use a generator to solve this instead?

Reply:
we could but I think it would have to be used indirectly since we don't have control over how this function is called. We certainly could do something like:
```
myCoolTests:
    runSetup()
    yield foo
    runTests

myCoolTestsGlobal = myCoolTests()

run_once_after_stage_opens():
    if not next(myCoolTests, None):
         unsubscribeThisFunction()
```
But that also seems funky. I can't think of a non-funky way of doing this since we can't use `wait_n_frames`

Reply:
Okay. I'm not going to block over this. I'll approve.

Reply:
I've got to fix a (superfluous?) import to get the windows build working anyway, so if there's a cleaner way to accomplish this I'm happy to implement it